26_CentralLimitBoxBlur/main.cpp

```diff
@@ -0,0 +1,432 @@
+				intendedSubmit, textureToBlur->getBuffer(), inCpuTexInfo.format,
+				inputGpuImg.get(), IImage::LAYOUT::TRANSFER_DST_OPTIMAL, textureToBlur->getRegions()
+			);
+			if( !uploaded )
+			{
+				return logFail( "Failed to upload cpu tex!\n" );
+			}
+
+			const IGPUCommandBuffer::SImageMemoryBarrier<IGPUCommandBuffer::SOwnershipTransferBarrier> releaseOwnership[] = {
+				{
+					.barrier = {
+						.dep = {
+							.srcStageMask = PIPELINE_STAGE_FLAGS::COPY_BIT, .srcAccessMask = ACCESS_FLAGS::TRANSFER_WRITE_BIT,
+							// dst flags are ignored upon a RELEASE
+						},
+						.ownershipOp = IGPUCommandBuffer::SOwnershipTransferBarrier::OWNERSHIP_OP::RELEASE,
+						.otherQueueFamilyIndex = getComputeQueue()->getFamilyIndex()
+					},
+					.image = inputGpuImg.get(),
+					.subresourceRange = {.aspectMask = IGPUImage::EAF_COLOR_BIT, .levelCount = 1, .layerCount = 1 },
+					.oldLayout = IGPUImage::LAYOUT::TRANSFER_DST_OPTIMAL,
+				    .newLayout = IGPUImage::LAYOUT::READ_ONLY_OPTIMAL,
+				}
+			};
+			if( !cmdbuf->pipelineBarrier( nbl::asset::EDF_NONE, { .imgBarriers = releaseOwnership } ) )
+			{
+				return logFail( "Failed to issue barrier!\n" );
+			}
+
+			cmdbuf->end();
+			const IQueue::SSubmitInfo info = intendedSubmit;
+			queue->submit({&info,1});
+			queue->endCapture();
+
+			// WARNING : Depending on OVerflows, `transferDone->value!=1` so if you want to sync the compute submit against that,
+			// use `transferDone` directly as the wait semaphore!
+			const ISemaphore::SWaitInfo waitInfo = {transferDone->semaphore,transferDone->value};
+			m_device->blockForSemaphores( { &waitInfo,1 } );
+		}
+		
+		constexpr size_t StartedValue = 0;
+		constexpr size_t FinishedValue = 45;
+		static_assert( StartedValue < FinishedValue );
+		smart_refctd_ptr<ISemaphore> progress = m_device->createSemaphore( StartedValue );
+		IQueue* queue = getComputeQueue();
+
+		smart_refctd_ptr<nbl::video::IGPUCommandBuffer> cmdbuf;
+		smart_refctd_ptr<nbl::video::IGPUCommandPool> cmdpool = m_device->createCommandPool(
+			queue->getFamilyIndex(), IGPUCommandPool::CREATE_FLAGS::RESET_COMMAND_BUFFER_BIT );
+		if( !cmdpool->createCommandBuffers( IGPUCommandPool::BUFFER_LEVEL::PRIMARY, 1u, &cmdbuf ) )
+		{
+			return logFail( "Failed to create Command Buffers!\n" );
+		}
+
+		struct packed_data_t
+		{
+			uint32_t direction : 2;
+			uint32_t channelCount : 3 = 3; // TODO: don't hardcode
+			uint32_t wrapMode : 2 = WrapMode::WRAP_MODE_CLAMP_TO_EDGE;
+			uint32_t borderColor : 3 = BorderColor::BORDER_COLOR_FLOAT_OPAQUE_BLACK;
+
+			explicit operator uint32_t() const {
+				return *reinterpret_cast< const uint32_t* >( this );
+			}
+		};
+
+		BoxBlurParams pushConstData = {
+			.inputDimensions = {0,0,0,uint32_t( packed_data_t{} )}, .chosenAxis = {1, 0}, .radius = 4.f
+		};
+
+		auto gpuTexSize = inputGpuImg->getCreationParameters().extent;
+		const uint32_t WorkgroupCount = ( gpuTexSize.width * gpuTexSize.height ) / WorkgroupSize;
```

Comment:
dispatches should be `(1,height,depth)` and `(width,1,depth)` for the x-axis and y-axis passess

you map the local invocations (whole workgroup) to your scanline, first x then y, and so on